entity-service-impl/src/main/java/org/hypertrace/entity/type/service/v2/EntityTypeServiceImpl.java

```diff
@@ -115,29 +117,37 @@ public void queryEntityTypes(
       org.hypertrace.entity.type.service.v2.QueryEntityTypesRequest request,
       io.grpc.stub.StreamObserver<org.hypertrace.entity.type.service.v2.QueryEntityTypesResponse>
           responseObserver) {
-    Optional<String> tenantId = RequestContext.CURRENT.get().getTenantId();
+    final RequestContext requestContext = RequestContext.CURRENT.get();
+    Optional<String> tenantId = requestContext.getTenantId();
     if (tenantId.isEmpty()) {
       responseObserver.onError(new ServiceException("Tenant id is missing in the request."));
       return;
     }
 
-    Iterator<Document> entityTypes =
-        entityTypeCollection.search(transform(tenantId.get(), request.getNameList()));
-
-    List<EntityType> entityTypeList = new ArrayList<>();
-    while (entityTypes.hasNext()) {
-      String json = entityTypes.next().toJson();
-      try {
-        entityTypeList.add(EntityTypeDocument.fromJson(json).toProto());
-      } catch (JsonProcessingException e) {
-        LOGGER.warn("Failed to parse the EntityType json: {}", json, e);
-        responseObserver.onError(new RuntimeException("Error querying the entity types."));
-        return;
+    try (final CloseableIterator<Document> entityTypes =
+        entityTypeCollection.search(transform(tenantId.get(), request.getNameList()))) {
+
+      List<EntityType> entityTypeList = new ArrayList<>();
+      while (entityTypes.hasNext()) {
+        String json = entityTypes.next().toJson();
+        try {
+          entityTypeList.add(EntityTypeDocument.fromJson(json).toProto());
+        } catch (JsonProcessingException e) {
+          LOGGER.warn("Failed to parse the EntityType json: {}", json, e);
+          responseObserver.onError(new RuntimeException("Error querying the entity types."));
+          return;
+        }
       }
+      responseObserver.onNext(
+          QueryEntityTypesResponse.newBuilder().addAllEntityType(entityTypeList).build());
+      responseObserver.onCompleted();
+    } catch (final Exception e) {
+      LOGGER.error("Unknown error occurred", e);
+      responseObserver.onError(
+          Status.INTERNAL
+              .withDescription("Unknown error occurred")
+              .asRuntimeException(requestContext.buildTrailers()));
```

Comment:
the trailers are no longer needed as of grpc utils `0.13.1` (not sure what this repo is using). Among other things, I updated the server interceptor used for error propgation to automatically add the request context used to make the call when returning. Only need to set it in the status if we want to override that or it wasn't used to originate the call (such as from a streaming app). https://github.com/hypertrace/java-grpc-utils/blob/1f94e38d46a2ac35b73c25cb9615920386daa4cc/grpc-server-utils/src/main/java/org/hypertrace/core/grpcutils/server/ThrowableResponseInterceptor.java#L39